src/birthder.js

```diff
@@ -80,96 +79,63 @@
 
     // Link together the specified birthday and user and store the link in the brain.
     robot.respond(routes.set, async (msg) => {
-      let date
-      let name
-      let user
-      let users
-
       if (!await routines.isAdmin(robot, msg.message.user.name.toString())) {
         msg.send(utils.MSG_PERMISSION_DENIED)
         return
       }
 
-      name = msg.match[2].trim()
-      date = msg.match[3]
-      users = []
-
-      for (const u of robot.brain.usersForFuzzyName(name)) {
-        if (await routines.isUserActive(robot, u)) {
-          users.push(u)
-        }
-      }
+      const name = msg.match[2].trim()
+      const date = msg.match[3]
+      const user = await routines.findUserByName(robot, name)
 
       if (!routines.isValidDate(date, utils.DATE_FORMAT)) {
         msg.send(utils.MSG_INVALID_DATE)
         return
       }
 
-      if (users.length === 1) {
-        user = users[0]
+      if (user) {
         user.dateOfBirth = date
 
         return msg.send(`Saving ${name}'s birthday.`)
-      } else if (users.length > 1) {
-        return msg.send(utils.getAmbiguousUserText(users))
       } else {
         return msg.send(`I have never met ${name}.`)
       }
     })
 
     // Print the users names whose birthdays match the specified date.
     robot.respond(routes.check, async (msg) => {
-      let date
-      let message
-      let users = []
-      let userNames
-
-      date = msg.match[2]
-
-      for (const u of utils.findUsersByDate(utils.BDAY_EVENT_TYPE, moment(date, utils.SHORT_DATE_FORMAT), robot.brain.data.users)) {
-        if (await routines.isUserActive(robot, u)) {
-          users.push(u)
-        }
-      }
+      const allUsers = await routines.getAllUsers(robot)
+      const date = moment(msg.match[2], utils.SHORT_DATE_FORMAT)
+      const users = utils.findUsersByDate(utils.BDAY_EVENT_TYPE, date, allUsers)
 
       if (users.length === 0) {
         return msg.send('Could not find any user with the specified birthday.')
       }
 
-      userNames = users.map(user => `@${user.name}`)
-      message = `${userNames.join(', ')}`
+      const userNames = users.map(user => `@${user.name}`)
+      const message = `${userNames.join(', ')}`
 
       return msg.send(message)
     })
 
     // Delete the birthday associated with the specified user name.
     robot.respond(routes.delete, async (msg) => {
-      let name = msg.match[2].trim()
-      let user
-      let users = []
-
-      if (!await routines.isAdmin(robot, msg.message.user.name.toString())) {
+      if (!await routines.isAdmin(robot, msg.message.user.name)) {
```

Comment:
This string looks like refactoring and not relative to the commit message refactoring.